csi/controller_server.go

```diff
@@ -97,19 +98,33 @@ func (cs *ControllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
 		volumeParameters["fromBackup"] = backup.Url
 	}
 
-	// check for already existing volume name
-	// ID and name are same in longhorn API
-	existVol, err := cs.apiClient.Volume.ById(req.GetName())
+	// Check for already existing volume name ID and name are same in longhorn API
+	// Ensure auto-corrected volume name not missed:
+	// - Convert to lower case of the definition of a label in DNS (RFC 1123)
+	// - Trim volume name of max name length limitation
+	volName := strings.ToLower(req.GetName())
+	if len(volName) > datastore.NameMaximumLength {
+		volName = strings.TrimRight(volName[:datastore.NameMaximumLength], "-")
```

Comment:
is it possible that 2 volumes with overlength names end up with the same truncated name?
If so then it seams that it would be possible that 2 pvc (full name) could both think that they are the owner of a single longhorn volume (truncated name)

Reply:
Great catch!
I've committed the fix in https://github.com/longhorn/longhorn-manager/pull/825/commits/009e20dd4e5a97671e77b2533a8d7a3a18c07ed6, this will replace the suffix of the new name with the first 8 characters of its full name checksum.